src/Filtering/Filter.coffee

```diff
@@ -168,6 +173,9 @@ Filter =
         $.addClass @nodes.root, hl...
     if noti and Unread.posts and (@ID > Unread.lastReadPost) and not QuoteYou.isYou(@)
       Unread.openNotification @, ' triggered a notification filter'
+    if watch and !ThreadWatcher.isWatched(@thread)
+      ThreadWatcher.add(@thread)
```

Comment:
This watches the thread if it sees any reply to the thread that matches the filter. Is that intentional?

Reply:
Yes. I figure they could have op: only to match starting posts.